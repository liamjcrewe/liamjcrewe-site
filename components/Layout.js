import styled from 'styled-components'

import Nav from './Nav'

const StyledLayout = styled.div`
  height: 100vh;
  padding: 20px;
  background-color: ${props => props.theme.colors.primary};
  color: ${props => props.theme.colors.secondary};
`

const Layout = props => (
  <StyledLayout>
    <Nav />
    {props.children}
  </StyledLayout>
)

export default Layout
