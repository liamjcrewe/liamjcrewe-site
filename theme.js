export default {
  colors: {
    primary: '#000000',
    secondary: '#ffffff',
    tertiary: '#08c4e6'
  }
}
