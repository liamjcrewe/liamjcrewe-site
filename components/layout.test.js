import { render } from '@testing-library/react'
import { ThemeProvider } from 'styled-components'

import 'jest-styled-components'

import theme from '../theme'
import Layout from './layout'

describe('Layout component', () => {
  it('renders itself and its children and matches snapshot', () => {
    const { container } = render(
      <ThemeProvider theme={theme}>
        <Layout>
          <span>Some content</span>
          <span>Some more content</span>
        </Layout>
      </ThemeProvider>
    )

    expect(container).toMatchSnapshot()
  })
})
