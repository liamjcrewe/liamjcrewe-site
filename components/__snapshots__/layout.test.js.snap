// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout component renders itself and its children and matches snapshot 1`] = `
.c0 {
  min-height: 100vh;
  color: #ffffff;
  background: url(/background.jpg);
  background-position: center;
  background-repeat: no-repeat;
  background-size: cover;
}

.c1 {
  min-height: inherit;
  background-color: rgba(0,0,0,0.5);
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <span>
        Some content
      </span>
      <span>
        Some more content
      </span>
    </div>
  </div>
</div>
`;
